DQM/CTPPS/plugins/CTPPSPixelDQMSource.cc

```diff
@@ -0,0 +1,462 @@
+       hRPotActivROCsMax[indexP]->getTH2F()->SetOption("colz");
+
+       int nbins = pixRowMAX/pixBinW;
+
+       for(int p=0; p<NplaneMAX; p++) {
+         sprintf(s,"plane_%d",p);
+         string pd = rpd+"/"+string(s);
+         ibooker.setCurrentFolder(pd);
+         string st1 = ": "+rpTitle+"_"+string(s);
+         st = "hits position";
+         h2xyHits[indexP][p]=ibooker.book2DD(st,st1+";pix col;pix row", 
+	   nbins,0,pixRowMAX,nbins,0,pixRowMAX);
+	 h2xyHits[indexP][p]->getTH2D()->SetOption("colz");
+
+ 	 st = "adc average value";
+	 hp2xyADC[indexP][p]=ibooker.bookProfile2D(st,st1+";pix col;pix row",
+	   nbins,0,pixRowMAX,nbins,0,pixRowMAX,100,0,1.e10,"");
+	 hp2xyADC[indexP][p]->getTProfile2D()->SetOption("colz");
+
+         st = "hits multiplicity";
+         hHitsMult[indexP][p]=ibooker.book1DD(st,st1+";number of hits;N / 1 hit",
+	   hitMultMAX,0,hitMultMAX);
+
+         ibooker.setCurrentFolder(pd + "/ROCs");
+         int index = getPlaneIndex(arm,stn,rp,p);
+
+         for(int roc=0; roc<6; roc++) {
+           sprintf(s,"hits in ROC_%d",roc);
+           string st = string(s);
+
+           h2xyROCHits[index][roc]=ibooker.book2DD(st,st1+";pix row;pix col",
+		ROCSizeInX,0,ROCSizeInX,ROCSizeInY,0,ROCSizeInY);
+           h2xyROCHits[index][roc]->getTH2D()->SetOption("colz");
+         }
+       } // end of for(int p=0; p<NplaneMAX;..
+
+     } // end for(int rp=0; rp<NRPotsMAX;...
+   } // end of for(int stn=0; stn<
+  } //end of for(int arm=0; arm<2;...
+
+ return;
+}
+
+//-------------------------------------------------------------------------------
+void CTPPSPixelDQMSource::beginLuminosityBlock(edm::LuminosityBlock const& lumiSeg, 
+                                            edm::EventSetup const& context) 
+{
+}
+//---------------------------------------------------------------------------------
+
+void CTPPSPixelDQMSource::analyze(edm::Event const& event, edm::EventSetup const& eventSetup)
+{
+  ++nEvents;
+ for(int ind=0; ind<2*3*NRPotsMAX; ind++) 
+   for(int p=0; p<NplaneMAX; p++) HitsMultPlane[ind][p] = 0;
+
+ for(int ind=0; ind<2*3*NRPotsMAX*NplaneMAX; ind++) 
+   for(int rp=0; rp<NROCsMAX; rp++) HitsMultROC[ind][rp] = 0;
+
+  Handle< DetSetVector<CTPPSPixelDigi> > pixDigi;
+  event.getByToken(tokenDigi, pixDigi);
+
+  hBX->Fill(event.bunchCrossing());
+  hBXshort->Fill(event.bunchCrossing());
+
+  bool valid = false;
+  valid |= pixDigi.isValid();
+//  valid |= Clus.isValid();
+
+ if(!valid && verbosity) LogPrint("CTPPSPixelDQMSource") <<"No valid data in Event "<<nEvents;
+
+if(pixDigi.isValid())
+  for(const auto &ds_digi : *pixDigi)
+  {
+   int idet = getDet(ds_digi.id);
+   if(idet != DetId::VeryForward) {
+     if(verbosity>1) LogPrint("CTPPSPixelDQMSource") <<"not CTPPS: ds_digi.id"<<ds_digi.id;
+     continue;
+   }
+//   int subdet = getSubdet(ds_digi.id);
+
+   int plane = getPixPlane(ds_digi.id);
+   int arm = CTPPSDetId(ds_digi.id).arm()&0x1;
```

Comment:
First, I would have expected the type here to be CTPPSPixelDetId?

Second, its probably better to create a CTPPS(Pixel)DetId object as use it instead of creating new ones each line